pagefaultgames/pokerogue
# [Bug] Fix Post-Summon abilities not applying after initially accepting a switch prompt

## What are the changes the user will see?
Currently, if the player agrees to switch a Pokemon at the start of a wave, but then cancels the switch in the party menu, the original Pokemon's post-summon abilities (e.g. Drought/Drizzle) do not apply. This fixes that.

## Why am I making these changes?
fixes https://github.com/pagefaultgames/pokerogue/issues/3482

## What are the changes from a developer perspective?
- Moved a call to remove a Pokemon's `PostSummonPhase` from the phase queue from the end of `CheckSwitchPhase` (i.e. when the player accepts the prompt to switch) to the end of `SwitchPhase` (i.e. after the player confirms the Pokemon to switch in from the party UI)
- Fixed some potential sequencing issues in `PostSummonPhase` that may cause Commander to activate after calling `PostSummonPhase.end()`.

### Screenshots/Videos

https://github.com/user-attachments/assets/946dc92f-26d1-4b3a-8756-bb269be93e55

## How to test the changes?
1. Start a run with a Pokemon with a post-summon ability as your first starter choice and at least 1 other Pokemon.
2. When prompted to switch, accept the initial prompt, then back out of the party menu.
3. Your Pokemon's post-summon effects should trigger after declining the switch.

## Checklist
